### ISSUE TITLE
minify function has bogus (private) documentation

### ISSUE THREAD

**Issue by lemire** [simdjson/simdjson#911]
Does not look right:

```C++
  /**
   * @private For internal implementation use
   *
   * Run a full document parse (ensure_capacity, stage1 and stage2).
   *
   * Overridden by each implementation.
   *
   * @param buf the json document to parse. *MUST* be allocated up to len + SIMDJSON_PADDING bytes.
   * @param len the length of the json document.
   * @param dst the buffer to write the minified document to. *MUST* be allocated up to len + SIMDJSON_PADDING bytes.
   * @param dst_len the number of bytes written. Output only.
   * @return the error code, or SUCCESS if there was no error.
   */
  WARN_UNUSED virtual error_code minify(const uint8_t *buf, size_t len, uint8_t *dst, size_t &dst_len) const noexcept = 0;
```

**Comment by jkeiser** [comment]
Indeed. I think it's just the first sentence that's wrong. While I think we don't need any padding for minify either, I'm hesitant to relax the requirement.

**Comment by jkeiser** [comment]
[Fixed in #927](https://github.com/simdjson/simdjson/commit/a6e4933d93f8cc79615533aa766e99dd314b8bd7#diff-eedeb40407c1a41e749cac8daf11d167).
